contracts/account/extensions/ERC7579Modules/SocialRecoveryExecutor.sol

```diff
@@ -0,0 +1,475 @@
+                                MODULE MAIN LOGIC
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Starts the recovery process for an account
+    /// @dev Requires threshold number of valid guardian signatures
+    /// @param account The account to start recovery for
+    /// @param guardianSignatures Array of guardian signatures authorizing the recovery
+    function startRecovery(
+        address account,
+        GuardianSignature[] calldata guardianSignatures
+    ) external whenRecoveryIsNotStarted(account) {
+        if (guardianSignatures.length < _recoveryConfigs[account].threshold) {
+            revert ThresholdNotMet();
+        }
+        if (!guardianSignaturesAreValid(account, guardianSignatures)) {
+            revert InvalidGuardianSignatures();
+        }
+
+        // increment nonce.
+        _recoveryConfigs[account].nonce++;
+
+        // set recovery start time.
+        _recoveryConfigs[account].recoveryStart = block.timestamp;
+
+        emit RecoveryStarted(account);
+    }
+
+    /// @notice Executes the recovery process after timelock period
+    /// @dev Can only execute calls to reconfigure installed Validator Modules
+    /// @param account The account to execute recovery for
+    /// @param guardianSignatures Array of GuardianSignature authorizing the recovery
+    /// @param executionCalldata The encoded call to reconfigure a Validator Module
+    function executeRecovery(
+        address account,
+        GuardianSignature[] calldata guardianSignatures,
+        bytes calldata executionCalldata
+    ) external whenRecoveryIsReady(account) nonReentrant {
+        if (guardianSignatures.length < _recoveryConfigs[account].threshold) {
+            revert ThresholdNotMet();
+        }
+        if (!guardianSignaturesAreValid(account, guardianSignatures)) {
+            revert InvalidGuardianSignatures();
+        }
+
+        // exection data should be at least:20 bytes for targetValidatorModule, 32 for value and 4 for recovery selector.
+        if (executionCalldata.length < 56) {
+            revert InvalidRecoveryCallData();
+        }
+
+        // @TBD sending value to the validator module.
+        // Rationale: In some very specific scenarios, the validator module might need to be paid for the recovery,
+        // or the recovery logic includes to take all the balance of the account.
+        (address targetValidatorModule, , ) = ERC7579Utils.decodeSingle(executionCalldata);
+
+        // check the target is an installed Validator Module on the account.
+        if (!IERC7579ModuleConfig(account).isModuleInstalled(MODULE_TYPE_VALIDATOR, targetValidatorModule, "")) {
+            revert InvalidInstalledValidatorModule();
+        }
+
+        // create single call mode
+        Mode mode = ERC7579Utils.encodeMode(
+            ERC7579Utils.CALLTYPE_SINGLE, // enforce single call
```

Comment:
Forcing single is not great. What if you want to do multiple operations during the recovery.

IMO the call should just be
```Address.functionCall(account, executionCalldata)```;

You may do some check that about the first 4 bytes of the executionCalldata (selector) ... to make sure it actually encodes a `executeFromExecutor` ... but honestly, what else is an executionModule going to call anyway ?